### PR TITLE
Team stats

### PR DESCRIPTION
**Pull request by araspaas**
finished team stats. added all tests and methods to stat tracker.

### CODE REVIEW

File: lib/team_stats.rb [hunk 1]
```diff
@@ -59,4 +59,34 @@ def fewest_goals_scored(team_id)
...
     x.goals
   end
+
+  def favorite_opponent(team_id)
+    find_by_given_team_id = @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
+    x = find_by_given_team_id.map {|game| game.game_id}
+    y = @game_teams.find_all {|game| x.include?(game.game_id) && game.team_id != team_id}
+    sorted_by_team_id = y.group_by {|game| game.team_id}
+    result = sorted_by_team_id.transform_values do |game|
+      w = game.find_all do |game|
```
Comment (ShaunDaneJames) [inline lib/team_stats.rb:69]: Seems like a good opportunity to refactor.

**Comment by robertorodriguez12** [comment]
Looking good!!! Great work on the methods here. Thank you, Austin.

**Review by robertorodriguez12 — APPROVED** [review]

**Review by ShaunDaneJames — APPROVED** [review]
Nice work Austin, looks great! Seems we are ready to start refactoring.

File: lib/team_stats.rb [hunk 2]
```diff
@@ -59,4 +59,34 @@ def fewest_goals_scored(team_id)
...
+    @teams.find {|team| team.team_id == q}.team_name
+  end
+
+  def rival(team_id)
+    find_by_given_team_id = @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
+    x = find_by_given_team_id.map {|game| game.game_id}
+    y = @game_teams.find_all {|game| x.include?(game.game_id) && game.team_id != team_id}
+    sorted_by_team_id = y.group_by {|game| game.team_id}
+    result = sorted_by_team_id.transform_values do |game|
+      w = game.find_all do |game|
```
Comment (dcoleman21) [inline lib/team_stats.rb:84]: Let's come up with other variable names here.

File: test/team_stats_test.rb [hunk 3]
```diff
@@ -14,9 +14,9 @@ class TeamStatsTest < Minitest::Test
 
   def setup
     data = {
-            games:      './data/games.csv',
-            teams:      './data/teams.csv',
-            game_teams: './data/game_teams.csv'
+            games:      './fixtures/games_fixture.csv',
```
Comment (dcoleman21) [inline test/team_stats_test.rb:17]: Nice pulling these from the fixture files

File: test/team_test.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@ class TeamTest < MiniTest::Test
 
   def setup
     team_param = {team_id: "1",
-                  franchise_id: "23",
+                  franchiseid: "23",
```
Comment (dcoleman21) [inline test/team_test.rb:9]: good catch on this naming convention

**Review by dcoleman21 — COMMENTED** [review]
Nice job changing these to the fixture files so that our speed will get better

**Pull request merged**